### DISCUSSION TITLE
Schema-Level Multi Tenancy for PostgreSQL Databases - Edge Case

### DISCUSSION THREAD

**Discussion by biyani701** [sqlalchemy/alembic#1496] [Usage Questions]
**In my models few tables need to be created in public schema and remaining tables in the specific schema**
Referencing https://alembic.sqlalchemy.org/en/latest/cookbook.html section "Rudimental Schema-Level Multi Tenancy for PostgreSQL Databases", in the first pass if i pass the schema first_schema, the tables are created 
1. in first_schema where model does not schema defined 
2. in public where schema = public is defined.

Now if i run the migration with second_schema, because the tables in public schema exists, the migration run fails,.

Is it possible to handle this by adding some hooks?

**Databases / Backends / Drivers targeted**

 postgresql+psycopg2

**Example Use**

**Additional context**

**Have a nice day!**

**Comment by CaselIT** [comment]
Hi,

I haven't tried this particular suggestion, but if I understand it correctly the tables in the public schema should not be detected when setting a schema that's no public, so in your first run with `first_schema` only table from first schema should be returned.

Could you provide your env.py?
